src/Material.cc

```diff
@@ -358,3 +366,64 @@ void Material::SetFilePath(const std::string &_filePath)
 {
   this->dataPtr->filePath = _filePath;
 }
+
+//////////////////////////////////////////////////
+ignition::math::Color Material::Implementation::GetColor(
+    const std::string &_colorType, Errors &_errors)
+{
+  ignition::math::Color color(0, 0, 0, 1);
+
+  if (!this->sdf->HasElement(_colorType))
+    return color;
+
+  std::string colorStr = this->sdf->GetElement(_colorType)
+                                  ->GetValue()
+                                  ->GetAsString();
+  if (colorStr.empty())
+    return color;
```

Comment:
Same comment here about a warning message and getting the default color

Reply:
I appended to `_errors` for this case f4633ca1011bd63afe05cb51928a5736a52fa09c